GitHub_Learning/Feature/Onboarding/Application/OnboardingController.swift

```diff
@@ -21,6 +23,41 @@ class OnboardingControllerImpl: BaseControllerImpl {
             notifyListenersAboutUpdate()
         }
     }
+    private let storeOnboardingFinishedFacade: StoreOnboardingFinishedFacade
+    private let loadOnboardingFinishedFacade: LoadOnboardingFinishedFacade
+    
+    init(
+        storeOnboardingFinishedFacade: StoreOnboardingFinishedFacade,
+        loadOnboardingFinishedFacade: LoadOnboardingFinishedFacade
+    ) {
+        self.storeOnboardingFinishedFacade = storeOnboardingFinishedFacade
+        self.loadOnboardingFinishedFacade = loadOnboardingFinishedFacade
+    }
+}
+
+extension OnboardingControllerImpl: OnboardingController {
+    func getOnboardingFinished(completion: @escaping ((Bool) -> Void)) {
+        loadOnboardingFinished(completion: completion)
+    }
+    
+    func storeOnboardingFinished(isFinished: Bool) {
+        storeOnboardingFinishedFacade.store(StoreOnboardingFinishedRequest(isFinished: isFinished)) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(response: _): print("Settings succesfully stored")
+            case .failure(error: let error): self.notifyListeners(about: error)
+            }
+        }
+    }
+    
+    private func loadOnboardingFinished(completion: @escaping ((Bool) -> Void)) {
+        loadOnboardingFinishedFacade.load(LoadOnboardingFinishedRequest()) { [weak self] result in
+            guard let self = self else { return }
+            self.handleResponse(result) { response in
```

Comment:
Nevim jestli se to tady uplne hodi. Todle neni response z internetu.

Reply:
To je jen switch na succes a failiure s nějakýma logama. Tak bych to akorát musel psaát růčo + bych přišel o ty logy za free.